### DISCUSSION TITLE
[Prettier Plugin] Variants of classes with multiple variants doesn't sort?

### DISCUSSION THREAD

**Discussion by LetsZiggy** [tailwindlabs/tailwindcss#10273] [General]
does `prettier-plugin-tailwindcss` sort variants (default variants)?
eg: `hover:sm:block` ---> `sm:hover:block`

i've tried searching in discord and github but can't find a definitive answer

i've also tried tailwindcss playground's tidy function with and without `variantOrder` in the config

**Comment by RobinMalfait** [comment]
Hey!

It does not right now because the order of variants can change the behaviour in certain case.
E.g.: https://play.tailwindcss.com/1w2kKJ4Rdc

The `variantOrder` setting is not used anymore.

**Reply by LetsZiggy** [reply]
ok. thanks for the example. i didn't know that.
is variant order sorting on the roadmap or should we set our own convention?

**Reply by adamwathan** [reply]
Not planned right now just because the order you put them in affects the behavior and sometimes both behaviours are valid choices, so if we sorted them it could break your design. Some of them are safe to sort for sure but figuring all that out is a bit of a big project unfortunately.

**Reply by LetsZiggy** [reply]
ok. thanks for the clarification
